tests/test-iso-kargs.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+set -xeuo pipefail
+
+fatal() {
+    echo "$@" >&2
+    exit 1
+}
+
+iso=$1; shift
+iso=$(realpath "${iso}")
+
+tmpd=$(mktemp -d)
+trap 'rm -rf "${tmpd}"' EXIT
+cd "${tmpd}"
+
+cp --reflink=auto "${iso}" "test.iso"
+iso=test.iso
+orig_hash=$(sha256sum "${iso}")
+
+# Sanity-check the ISO doesn't somehow already have the karg we're testing with.
+if coreos-installer iso kargs show "${iso}" | grep -q foobar; then
+    fatal "Unexpected foobar karg in iso kargs"
+fi
+
+# Test all the modification operations.
+coreos-installer iso kargs modify -a foobar=oldval "${iso}"
+coreos-installer iso kargs show "${iso}" | grep -q foobar=oldval
+coreos-installer iso kargs modify -a dodo "${iso}"
+coreos-installer iso kargs show "${iso}" | grep -q 'foobar=oldval dodo'
+coreos-installer iso kargs modify -r foobar=oldval=newval "${iso}"
+coreos-installer iso kargs show "${iso}" | grep -q 'foobar=newval dodo'
+coreos-installer iso kargs modify -d foobar=newval "${iso}"
+if coreos-installer iso kargs show "${iso}" | grep -q 'foobar'; then
+    fatal "Unexpected foobar karg in iso kargs"
+fi
+coreos-installer iso kargs show "${iso}" | grep -q 'dodo'
+coreos-installer iso kargs modify -d dodo "${iso}"
+
+hash=$(sha256sum "${iso}")
+if [ "${orig_hash}" != "${hash}" ]; then
+    fatal "Hash doesn't match original hash: ${hash} vs ${orig_hash}"
+fi
+
+# Test the largest karg; this technically is a detail of the padding in the
+# distro configs. Here we're assuming an FCOS/RHCOS ISO: 1024 - 2 (2 for the
+# final '\n#' to put "COREOS_KARG_EMBED_AREA on the next line).
+long_karg=$(printf '%*s' 1022 | tr ' ' "k")
+coreos-installer iso kargs modify -a "${long_karg}" "${iso}"
+coreos-installer iso kargs show "${iso}" | grep -q ' k\+$'
+long_karg=$(printf '%*s' 1023 | tr ' ' "k")
+if coreos-installer iso kargs modify -a "${long_karg}" "${iso}" 2>err.txt; then
```

Comment:
This doesn't prove much unless we `reset` first.

Reply:
Good catch!